File: bcmc-base/src/main/java/com/adyen/checkout/bcmc/data/BcmcOutputData.java
/*
 * Copyright (c) 2019 Adyen N.V.
 *
 * This file is open source and available under the MIT license. See the LICENSE file for more info.
 *
 * Created by arman on 18/9/2019.
 */

package com.adyen.checkout.bcmc.data;

import android.support.annotation.NonNull;

import com.adyen.checkout.base.component.OutputData;
import com.adyen.checkout.base.validation.ValidatedField;
import com.adyen.checkout.card.data.ExpiryDate;

public final class BcmcOutputData implements OutputData {

    private final ValidatedField<String> mCardNumberField;
    private final ValidatedField<ExpiryDate> mExpiryDateField;

    public BcmcOutputData(
            @NonNull ValidatedField<String> cardNumberField,
            @NonNull ValidatedField<ExpiryDate> expiryDateField
    ) {
        mCardNumberField = cardNumberField;
        mExpiryDateField = expiryDateField;
    }

    @NonNull
    public ValidatedField<String> getCardNumberField() {
        return mCardNumberField;
    }

    @NonNull
    public ValidatedField<ExpiryDate> getExpiryDateField() {
        return mExpiryDateField;
    }

    @Override
    public boolean isValid() {
        return mCardNumberField.isValid()
                && mExpiryDateField.isValid();
    }
}

File: bcmc/src/main/java/com/adyen/checkout/bcmc/internal/ui/model/BcmcOutputData.kt
/*
 * Copyright (c) 2023 Adyen N.V.
 *
 * This file is open source and available under the MIT license. See the LICENSE file for more info.
 *
 * Created by oscars on 15/2/2023.
 */
package com.adyen.checkout.bcmc.internal.ui.model

import com.adyen.checkout.card.internal.ui.model.ExpiryDate
import com.adyen.checkout.components.core.internal.ui.model.FieldState
import com.adyen.checkout.components.core.internal.ui.model.OutputData

internal data class BcmcOutputData internal constructor(
    val cardNumberField: FieldState<String>,
    val expiryDateField: FieldState<ExpiryDate>,
    val cardHolderNameField: FieldState<String>,
    val isStoredPaymentMethodEnabled: Boolean
) : OutputData {
    override val isValid: Boolean
        get() = (
            cardNumberField.validation.isValid() &&
                expiryDateField.validation.isValid() &&
                cardHolderNameField.validation.isValid()
            )
}
